compiler/luci-interpreter/include/luci_interpreter/Interpreter.h

```diff
@@ -30,8 +30,14 @@
 namespace luci_interpreter
 {
 
-class KernelMap;
-class TensorMap;
+class ExecutionObserver
+{
+public:
+  virtual ~ExecutionObserver();
+
+  // Called when the value of a tensor has changed during execution.
```

Comment:
Maybe "// Called after each operator is executed" ? Because some operators may not change the value of a tensor.

How about using the name`postOperatorExecution` instead of `onTensorChanged` ?

Reply:
It is also called for Input and Const nodes, which are not actually executed.
I prefer to think of it as of an event of updating tensor, rather than an event of finishing execution of an operation.
But we can also add preOperatorExecution/postOperatorExecution for operation even to s if they could be helpful for some purposes (measuring time, for example).

> Because some operators may not change the value of a tensor.

I couldn't invent a better name considering the above. postTensorWrite, may be? I'm open o other suggestions :)

Reply:
Also note that for operators with multiple outputs this method will be called for each output tensor (I still hope we will get adequate support for multiple outputs someday).
On the contrary, posrOperatorExecution will be called just once for each operator.

Reply:
> I prefer to think of it as of an event of updating tensor, rather than an event of finishing execution of an operation.

I see. Then postTensorWrite looks good.

Reply:
> I see. Then postTensorWrite looks good.

Fixed